### ISSUE TITLE
Error on BigQueryIntegrationTest

### ISSUE THREAD

**Issue by PedroMDuarte** [mozilla/gcp-ingestion#489]
Hi All, I was running the integration test and ran into the following error:
```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running com.mozilla.telemetry.integration.BigQueryIntegrationTest
[ERROR] Tests run: 4, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 27.174 s <<< FAILURE! - in com.mozilla.telemetry.integration.BigQueryIntegrationTest
[ERROR] canWriteViaFileLoads(com.mozilla.telemetry.integration.BigQueryIntegrationTest)  Time elapsed: 11.159 s  <<< ERROR!
org.apache.beam.sdk.Pipeline$PipelineExecutionException: java.io.IOException: Failed to close some writers
	at com.mozilla.telemetry.integration.BigQueryIntegrationTest.canWriteViaFileLoads(BigQueryIntegrationTest.java:171)
Caused by: java.io.IOException: Failed to close some writers
```
The commit where I ran this from is:
```
55a494e (HEAD -> master, origin/master, origin/HEAD) Bump psutil from 5.5.1 to 5.6.0 in /ingestion-edge
```

Wondering if anyone has any clues as to why this test might fail. 

Btw, thank you for open sourcing this work! I have been learning a lot about Beam by studying your codebase.

**Comment by jklukas** [comment]
Hi @PedroMDuarte! I'm really glad to hear reading the codebase here has been useful. How did you find it? Did it just happen to show up while searching for beam-related things?

That error doesn't sound familiar to me. My best guess is that it's permissions related. Were you running this using service account creds?

Can you dump a full stack trace? `mvn test -DtrimStackTrace=false`

**Comment by PedroMDuarte** [comment]
Hi @jklukas yes I found it while searching for a way to do proper routing of errors in beam. I really like the approach you all have taken. 

I will close this issue as it might be outdated by now, but I'll take note of the full stack trace option and will open up a new issue later on if I can reproduce it with the full stack trace.

**Comment by jklukas** [comment]
We contributed an exception handling interface for MapElements and FlatMapElements that will be available in Beam 2.12 to be released in the next week or two, so that may be of interest to you. It will allow us to remove some of our custom classes here.

**Comment by PedroMDuarte** [comment]
That's interesting. Thanks! I'll take a look.
